File: python-algorithm/leetcode/problem_820.py
"""820. Short Encoding of Words
https://leetcode.com/problems/short-encoding-of-words/
"""
from typing import List

"""
Input: words = ["time", "me", "bell"] Output: 10
Explanation: A valid encoding would be s =
"time#bell#" and indices = [0, 2, 5]
"""
class Solution:
    def minimumLengthEncoding(self, words: List[str]) -> int:
        ans = words[0] + '#'
        for i in range(1, len(words)):
            word = words[i]
            
        return 0

File: java-algorithm/src/main/java/com/leetcode/Problem820.java
package com.leetcode;

/**
 * 820. Short Encoding of Words
 * https://leetcode.com/problems/short-encoding-of-words/
 */
public class Problem820 {
    public int minimumLengthEncoding(String[] words) {
        return 0;
    }

    private static class TrieNode {
        TrieNode[] children;
        
        public TrieNode() {
            this.children = new TrieNode[27];
        }

        public void insert(String word) {
            TrieNode cur = this;
            for(int i = 0; i < word.length(); i++) {
                int x = word.charAt(i) - 'a';
                if (cur.children[x] == null) {
                    cur.children[x] = new TrieNode();
                }
                cur = cur.children[x];
            }
        }
    }
}

